drivers/hwinfo/hwinfo_stm32.c

```diff
@@ -15,8 +15,27 @@
 #include <string.h>
 #include <zephyr/sys/byteorder.h>
 
+#define STM32_UID_SIZE_IN_WORDS	3
+
+#if defined(CONFIG_SOC_SERIES_STM32MP13X)
+
+/* No ll_utils for the stm32mp13x series, instead use the HAL functions */
+#define STM32_UID_WORD_0 HAL_GetUIDw2()
+#define STM32_UID_WORD_1 HAL_GetUIDw1()
+#define STM32_UID_WORD_2 HAL_GetUIDw0()
+
+#else
+
+#define STM32_UID_WORD_0 LL_GetUID_Word2()
+#define STM32_UID_WORD_1 LL_GetUID_Word1()
+#define STM32_UID_WORD_2 LL_GetUID_Word0()
+
+#endif /* CONFIG_SOC_SERIES_STM32MP13X */
+
+#define SET_ID_X(x, _) dev_id.id[x] = STM32_UID_WORD_##x
```

Comment:
We are loosing `sys_cpu_to_be32` here, but I think we need to keep it based on the API documentation:
```
 * Drivers are responsible for ensuring that the ID data structure is a
 * sequence of bytes.  The returned ID value is not supposed to be interpreted
 * based on vendor-specific assumptions of byte order. It should express the
 * identifier as a raw byte sequence, doing any endian conversion necessary so
 * that a hex representation of the bytes produces the intended serial number.
```

cc @erwango what do you think?

If we have to keep it:
```suggestion
#define SET_ID_X(x, _) dev_id.id[x] = sys_cpu_to_be32(STM32_UID_WORD_##x)
```